internal/authentication/ldap_user_provider.go

```diff
@@ -197,19 +231,25 @@ type ldapUserProfile struct {
 	Username    string
 }
 
-func (p *LDAPUserProvider) resolveUsersFilter(userFilter string, inputUsername string) string {
-	inputUsername = p.ldapEscape(inputUsername)
+func (p *LDAPUserProvider) resolveUsersFilter(inputUsername string) (userFilter string) {
+	userFilter = p.configuration.UsersFilter
 
-	// The {input} placeholder is replaced by the users username input.
-	userFilter = strings.ReplaceAll(userFilter, "{input}", inputUsername)
+	if p.usersFilterReplacementInput {
+		// The {input} placeholder is replaced by the users username input.
+		userFilter = strings.ReplaceAll(userFilter, "{input}", p.ldapEscape(inputUsername))
+	}
+
+	if p.usersFilterReplacementEpochWin32 {
+		userFilter = strings.ReplaceAll(userFilter, "{epoch:win32}", strconv.FormatUint(utils.Win32EpochNow(), 10))
+	}
```

Comment:
gobench indicates this adds at most 1ns to this function, but potentially saves 100-150ns per check for instances where the replacement doesn't exist.